xarray/backends/api.py

```diff
@@ -1277,28 +1277,31 @@ def _validate_region(ds, region):
         )
 
 
-def _validate_datatypes_for_zarr_append(dataset):
-    """DataArray.name and Dataset keys must be a string or None"""
+def _validate_datatypes_for_zarr_append(zstore, dataset):
+    """If variable exists in the store, confirm dtype of the data to append is compatible with
+    existing dtype.
+    """
+
+    existing_vars = zstore.get_variables()
 
-    def check_dtype(var):
+    def check_dtype(vname, var):
         if (
-            not np.issubdtype(var.dtype, np.number)
-            and not np.issubdtype(var.dtype, np.datetime64)
-            and not np.issubdtype(var.dtype, np.bool_)
-            and not coding.strings.is_unicode_dtype(var.dtype)
-            and not var.dtype == object
+            vname not in existing_vars
+            or np.issubdtype(var.dtype, np.number)
+            or np.issubdtype(var.dtype, np.datetime64)
+            or np.issubdtype(var.dtype, np.bool_)
+            or var.dtype == object
         ):
-            # and not re.match('^bytes[1-9]+$', var.dtype.name)):
+            pass
```

Comment:
could you add a brief comment (e.g, based on your comment here: https://github.com/pydata/xarray/pull/6476#issuecomment-1097408858) to summarize why it's OK not to check these cases?

Reply:
Thanks so much for the review!

I've added the requested comment in  https://github.com/pydata/xarray/pull/6476/commits/9f97f00636d09454d6b7d2e13fa0775ed7770769